server.go

```diff
@@ -956,10 +970,23 @@ func (s *server) Start() error {
 		return nil
 	}
 
+	// If one sub system fails to start, the following code ensures that the
+	// previous started ones are stopped. It also ensures a proper wallet
+	// shutdown which is important for releasing its resources (boltdb, etc...)
+	cleanup := cleaner{
+		s.cc.chainView.Stop,
+		s.cc.feeEstimator.Stop,
+		func() error {
```

Comment:
does this has to be wrapped in a `func` like this?

Reply:
good catch! fixed